config/index.ts

```diff
@@ -0,0 +1,12 @@
+import * as nconf from 'nconf';
+
+const Config = () => {
+    nconf.argv().env("_");
+    const environment = nconf.get("NODE:ENV") || "development";
+    nconf.file(environment, "config/" + environment + ".json");
+    nconf.file("default", "config/config.json");
+};
+
+Config.prototype.get = (key) => nconf.get(key);
+
+export default new Config();
```

Comment:
dont export your config! once u set it it will work everywhere.  Call this file first in your index.ts so the config is the first thing you init (if you dont understand leave it for now)